data-r2dbc/src/main/java/io/micronaut/data/r2dbc/operations/DefaultR2dbcRepositoryOperations.java

```diff
@@ -483,11 +483,6 @@ public <T, R> Flux<R> findAll(@NonNull PreparedQuery<T, R> pq) {
 
                 SqlTypeMapper<Row, R> mapper = createMapper(preparedQuery, Row.class);
                 if (mapper instanceof SqlResultEntityTypeMapper<Row, R> entityTypeMapper) {
```

Comment:
Without this, child records are not processed properly. Removing this, we have the same behavior as in JDBC repository.

Reply:
Why is this needed? How can we have child records without joins?

Reply:
This query is produced
```
SELECT student_.`id`,student_.`version`,student_.`name`,student_.`creation_time`,student_.`last_updated_time`,student_books_.`id` AS books_id,student_books_.`author_id` AS books_author_id,student_books_.
`genre_id` AS books_genre_id,student_books_.`title` AS books_title,student_books_.`total_pages` AS books_total_pages,student_books_.`publisher_id` AS books_publisher_id,student_books_.
`last_updated` AS books_last_updated FROM `student` student_ INNER JOIN `book_student` student_books_book_student_ ON student_.`id`=student_books_book_student_.`student_id`
  INNER JOIN `book` student_books_ ON student_books_book_student_.`book_id`=student_books_.`id` WHERE (student_books_.`title` IN (?))
```
and it returns 5 records as we have 2 students and first student has 3 books and other 2 books. JdbcRepository reads this properly into 2 student records (does not populate books because there is no `@Join("books")` on `findAll(QuerySpecification querySpec)`. However, R2dbc returns 5 records until this piece of code is removed.
This is how query spec looks to generate this query:
```
static QuerySpecification<Student> byBookTitles(List<String> bookTitles) {
        return (root, query, criteriaBuilder) -> {
            boolean isCountQuery = query.getResultType().equals(Long.class);
            if (isCountQuery) {
                query.distinct(true);
            }
            SetJoin<Student, Book> join = root.joinSet("books", JoinType.LEFT);
            return join.get("title").in(bookTitles);
        };
    }
```

Reply:
It looks like the joins collection should not be empty

Reply:
It is, until adding `@Join("books")` on the `findAll(QuerySpecification method)` or on repository. It defines how records would be read I guees. Same for Jdbc and R2dbc repository operations.
Not sure if this is new behavior after recent refactoring or can be replicated in 3.x

Reply:
I'm confused. Why do we need to remove this check if the collection is not empty? This means the deleted check is skipped anyway. 
Or is it empty if the join is added using the criteria? Then it should be corrected and the fetch paths should be returned not empty.

Reply:
I will try to replicate this in master and try to show differences between jdbc and r2dbc handling.

Reply:
There is difference between Jdbc and R2dbc operations handling result. Here is Jdbc: 
```
SqlTypeMapper<ResultSet, R> mapper = createMapper(sqlStoredQuery, ResultSet.class);
        List<R> result;
        if (mapper instanceof SqlResultEntityTypeMapper<ResultSet, R> entityTypeMapper) {
            SqlResultEntityTypeMapper.PushingMapper<ResultSet, List<R>> manyMapper = entityTypeMapper.readAllWithJoins();
            while (rs.next()) {
                manyMapper.processRow(rs);
            }
            result = manyMapper.getResult();
            if (result == null) {
                result = List.of();
            }
        } else {
            result = new ArrayList<>();
            while (rs.next()) {
                result.add(
                    mapper.map(rs, sqlStoredQuery.getResultType())
                );
            }
        }
```
so it uses `entityTypeMapper.readAllWithJoins()` without checking if there are joins vs R2dbc (and both create mapper the same way)
```
SqlTypeMapper<Row, R> mapper = createMapper(preparedQuery, Row.class);
                if (mapper instanceof SqlResultEntityTypeMapper<Row, R> entityTypeMapper) {
                    final boolean hasJoins = !preparedQuery.getJoinFetchPaths().isEmpty();
                    if (!hasJoins) {
                        // Every row represents the entity record, we can return it directly
                        return executeAndMapEachRow(statement, entityTypeMapper::readEntity);
                    }
                    SqlResultEntityTypeMapper.PushingMapper<Row, List<R>> rowsMapper = entityTypeMapper.readAllWithJoins();
                    return executeAndMapEachRow(statement, row -> {
                        rowsMapper.processRow(row);
                        return "";
                    }).collectList().flatMapIterable(ignore -> rowsMapper.getResult());
                }
```
and this is my test in Jdbc has not failed but in R2dbc has failed. So, this is why I thought it is safe to remove these lines and to have same behavior in Jdbc and R2dbc

Reply:
I see, we probably don't need it for `findAll` methods. Is it going to be a problem for `findOne`?

Reply:
findOne seems ok, and also both Jdbc and R2dbc implementations work the same way.